docs/ising_qaoa.rst

```diff
@@ -0,0 +1,268 @@
+
+This wrapper is particulary useful for people that have been working with quantum annealers
+in the past. However, in comparison to current quantum annealers the Ising QAOA wrapper
+supports not only 2-local but also k-local interaction terms and the driver (mixer)
+Hamiltonian is not dictated by the hardware but can be defined by the user.
+
+For each Ising instance the user specifies the bias terms \\( h_{i} \\), the
+interaction tensor \\( J_{i,j,..,k} \\) and the approximation order of the algorithm.
+``ising_qaoa.py`` contains routines for approximating the ground state of
+the specified Ising-type Hamiltonian through the use of QAOA which itself
+finds the optimal rotation angles via Grove's
+`variational-quantum-eigensolver method <http://grove-docs.readthedocs.io/en/latest/vqe.html>`_.
+
+.. _quickstart-example:
+
+Quickstart Example
+-------------------
+
+To test your installation and get going we can run Ising QAOA for
+a very simple Ising Hamiltonian: the 2D checkerboard.
+(for a detailed explanation see :ref:`2d-checkerboard`)
+In your python script import the packages and connect to your QVM:
+
+.. code-block:: python
+
+    import pyquil.api as api
+    from grove.ising.ising_qaoa import ising_qaoa
+    qvm_connection = api.QVMConnection()
+
+Next we define the appropriate bias and interaction terms of the Ising Hamiltonian
+for the checkerboard problem:
+
+.. code-block:: python
+
+    J = {(0, 1): 1, (0, 2): 1, (1, 3): 1, (2, 3): 1}
+    h = {0: -1, 1: 1, 2: 1, 3: -1}
+
+There are plenty of optional configuration parameters for the algorithm but the two
+most important are the number of steps to use for the trotterization (roughly corresponds to
+the accuracy of the optimization) and the driver Hamiltonian (which determines how the
+state space is searched). We instantiate the algorithm and run the optimization routine on our QVM:
+
+.. code-block:: python
+
+    steps = 2
+    solution_string, ising_energy, _  = ising_qaoa(h=h, J=J, num_steps=steps)
+    print(f'The algorithm returned {solution_string} with an energy of {ising_energy}')
+
+When running this routine you should observe the expectation value converging towards -8.0
+and the solution with the highest probability should be \\( [1, -1, -1, 1] \\) with an energy of \\( -8.0 \\).
+
+We can verify this by running the Ising QAOA multiple times and collecting statistics.
+To do this, replace the last line of the last code block with:
+
+.. code-block:: python
+
+    runs = 10
+    stats = dict()
+    for _ in range(runs):
+        solution_string, ising_energy, _  = ising_qaoa(h=h, J=J, num_steps=steps)
+        if tuple(solution_string) in stats.keys():
+            stats[tuple(solution_string)] += 1
+        else:
+            stats[tuple(solution_string)] = 1
+    print(f'Solution statistics: {stats}')
+
+You should see that the algorithm returns the aforementioned solution with ~99% probability (unless the classical minimizer got stuck in a local minima).
```

Comment:
I don't see this when I run the routine - has something changed in Grove that breaks this example since you made this PR?

Reply:
@ampolloreno the problem was the declaration of `driver_operators` as an empty list before checking if it `is None` which resulted in no driver Hamiltonian. This has been resolved in commit [fc381d4](https://github.com/rigetticomputing/grove/pull/141/commits/fc381d4840a6325c2577ea5c48901b47769fe63a).

Reply:
Awesome, I'll check it out!